### PR TITLE
Feat/spi nand flash mt29f1 (IEC-225)

### PR DESCRIPTION
**Pull request by Teesmo**
# Checklist

- [ ] Component contains License
- [ ] Component contains README.md
- [ ] Component contains idf_component.yml file with `url` field defined
- [ ] Component was added to [upload job](https://github.com/espressif/idf-extra-components/blob/master/.github/workflows/upload_component.yml#L18)
- [ ] Component was added to [build job](https://github.com/espressif/idf-extra-components/blob/master/test_app/CMakeLists.txt#L8)
- [ ] _Optional:_ Component contains unit tests
- [ ] CI passing

# Change description
Added support for MT29F1G01ABAFDSF-AAT:F chip

### CODE REVIEW

File: spi_nand_flash/src/nand.c [hunk 1]
```diff
@@ -139,15 +139,21 @@ static esp_err_t spi_nand_micron_init(spi_nand_flash_device_t *dev)
...
     switch (device_id) {
     case MICRON_DI_34:
+        dev->read_page_delay_us = 115;
+        dev->program_page_delay_us = 240;
         dev->dhara_nand.num_blocks = 2048;
         dev->dhara_nand.log2_ppb = 6;        // 64 pages per block
         dev->dhara_nand.log2_page_size = 12; // 4096 bytes per page
         break;
+    case MICRON_DI_14:
+    case MICRON_DI_15:
```
Comment (igrr) [inline spi_nand_flash/src/nand.c:152]: Do you know what is the difference between these two device IDs?

Comment (Teesmo) [reply spi_nand_flash/src/nand.c:152]: It's operating voltage. The chip can run on two voltage levels: device ID 14 - 3.3V, device ID 15 -1.8V

Comment (igrr) [reply spi_nand_flash/src/nand.c:152]: Given that ESP chips only have 3.3V IOs, should the device ID 15 be excluded?

Comment (Teesmo) [reply spi_nand_flash/src/nand.c:152]: We could leave it out, but I don't see any harm in keeping it (some people may design parts of their circuits around 1.8V and decide to supply their storaage from that as well).

**Comment by Teesmo** [comment]
@igrr how far are we with this pull request?

**Comment by igrr** [comment]
@Teesmo Could you check the CLA (https://github.com/espressif/idf-extra-components/pull/432#issuecomment-2416980913)? It seems you made a commit with a different email (ending with `@gmail.co`) than the one added to your Github profile, so CLA Assistant doesn't recognize your sign-off.

Also, there have been some changes to spi_nand_flash.c in the meantime, could you please rebase the PR upon the latest master branch?

(If you have trouble doing this, just reply that you agree with the CLA and I can take care of rebasing and merging.)

**Comment by Teesmo** [comment]
I agree with the CLA. I've edited the email. I hope it checks off this time

**Review by igrr — APPROVED** [review]

**Pull request merged**